app/user.go

```diff
@@ -2383,8 +2383,35 @@ func (a *App) GetThreadsForUser(userId string, options model.GetUserThreadsOpts)
 	return threads, nil
 }
 
-func (a *App) UpdateThreadsReadForUser(userId string, timestamp int64) *model.AppError {
-	nErr := a.Srv().Store.Thread().MarkAllAsRead(userId, timestamp)
+func (a *App) UpdateThreadsReadForUser(userId, teamId string, timestamp int64) *model.AppError {
+	user, err := a.GetUser(userId)
+	if err != nil {
+		return err
+	}
+	var nErr error
+
+	memberships, nErr := a.Srv().Store.Thread().GetMembershipsForUser(userId, teamId)
```

Comment:
What is the purpose of this new block? Looks like pretty heavy, it is related with Unreads?